util/stopwaiter/stopwaiter.go

```diff
@@ -160,11 +160,33 @@ func (s *StopWaiterSafe) LaunchUntrackedThread(foo func()) {
 	go foo()
 }
 
-// call function iteratively in a thread.
+// CallIteratively calls function iteratively in a thread.
 // input param return value is how long to wait before next invocation
 func (s *StopWaiterSafe) CallIteratively(foo func(context.Context) time.Duration) error {
+	return s.CallIterativelyWithOverride(foo, nil)
+}
+
+// CallIterativelyWithOverride calls function iteratively in a thread.
+// The return value of foo is how long to wait before next invocation
+// Anything sent to overrideChan parameter triggers call to happen immediately
+func (s *StopWaiterSafe) CallIterativelyWithOverride(foo func(context.Context) time.Duration, overrideChan chan interface{}) error {
 	return s.LaunchThread(func(ctx context.Context) {
 		for {
+			done := 100
+			for done > 0 {
```

Comment:
1. I'm not certain about emptying the queue here. e.g. if a trigger was sent while the previous was processed, we probably do want to re-trigger immediately. I think that correct operation shouldn't require clearing here, and instead make sure that the channel is either not-buffered or has buffer of size 1 and that we use select when sending new triggers to not get stuck in case channel is already full.
2. If you do want to empty the queue - you can tell how many items exist in a channel using len(overrideChan), that will tell you how many max items you need to clear (we definitely shouldn't clear new items added while we're in the clearing loop)

Reply:
:+1: